File: Creational/FactoryMethod/TypeScript/Structure/ConcreteCreatorA.ts
import { Creator } from "./Creator";
import { ConcreteProductA } from "./ConcreteProductA";

/**
 * ConcreteCreatorA - Concrete implementation of Creator
 * Creates ConcreteProductA instances
 */
export class ConcreteCreatorA extends Creator {
  factoryMethod(): ConcreteProductA {
    return new ConcreteProductA();
  }
}

File: Creational/FactoryMethod/Java/Structure/ConcreteCreatorA.java
/**
 * ConcreteCreatorA - implementation cụ thể của Creator
 * Override factory method để tạo ConcreteProductA
 */
public class ConcreteCreatorA extends Creator {
    
    @Override
    public Product factoryMethod() {
        System.out.println("ConcreteCreatorA: Tạo Product A");
        return new ConcreteProductA();
    }
}
